proxy/envoy/config.go

```diff
@@ -303,9 +303,11 @@ func buildInboundFilters(instances []*model.ServiceInstance) (HTTPRouteConfigs,
 			http.VirtualHosts = append(http.VirtualHosts, host)
 		case model.ProtocolTCP:
 			cluster := buildInboundCluster(service.Hostname, endpoint.Port, port.Protocol)
-			route := buildTCPRoute(cluster, endpoint.Address, endpoint.Port)
 			config := tcpConfigs.EnsurePort(port.Port)
-			config.Routes = append(config.Routes, route)
+			config.Routes = append(config.Routes,
+				buildTCPRoute(cluster, []string{endpoint.Address}, endpoint.Port),
+				buildTCPRoute(cluster, []string{service.Address, endpoint.Address}, port.Port),
```

Comment:
Without that combination the route from pod B to "http://a:{90,9090}" didn't work. Pod B's kubeproxy/iptables re-routes HTTP request to Pod A's endpoint IP:Port but the listener config (below) doesn't have a route with matching port for the listener port. Maybe there is any issue with how we're binding endpoint to services in controller.go?

```
"port": 90,
        "filters": [
          {
            "type": "read",
            "name": "tcp_proxy",
            "config": {
              "stat_prefix": "tcp",
              "route_config": {
                "routes": [
                  {
                    "cluster": "inbound:9090",
                    "destination_ip_list": [
                      "10.16.0.218/32"			       // endpoint address
                    ],
                    "destination_ports": "9090" // port doesn't match listener port so this route is never used.
                  },
                  {
                    "cluster": "inbound:9090",
                    "destination_ip_list": [
                      "10.16.0.218/32"			// endpoint address
                    ],		
                    "destination_ports": "90" 		 
                  },
```